README.md

```diff
@@ -30,105 +48,6 @@ Get more information about this tool in the [contributing][] document.
 [report]: https://hyper.rs/h3/ci/compliance/report.html#/
 [contributing]: CONTRIBUTING.md
 
-## Features
-
-* HTTP/3 client and server implementation
-* Async only API
-* QUIC transport abstraction via traits in the [`quic`](./h3/src/quic.rs) module
-* Supported QUIC implementations to date are
-  [Quinn](https://github.com/quinn-rs/quinn) ([h3-quinn](./h3-quinn/))
-  and [s2n-quic](https://github.com/aws/s2n-quic)
-  ([s2n-quic-h3](https://github.com/aws/s2n-quic/tree/main/quic/s2n-quic-h3))
-
-## Overview
-
-* **h3** HTTP/3 implementation
-* **h3-quinn** QUIC transport implementation based on [Quinn](https://github.com/quinn-rs/quinn/)
-
-## Getting Started
-
-The [examples](./examples) directory can help get started in two ways:
-
-- There are ready-to-use `client` and `server` binaries to interact with _other_ HTTP/3 peers. Check the README in that directory.
-- The source code of those examples can help teach how to use `h3` as either a client or a server.
-
-### Server
```

Comment:
I\`m not sure if we should leave the small examples in the readme.

Reply:
The code to be pasted here would be snippets from the examples anyway. I assume a pointer to the examples should be enough to get people started right?

Reply:
I\`m absolutely not sure about this one. 
On the one hand this does\`t get tested or compiled so when api changes we have to remember to update this and the information provided from this is also in the examples and the docs.
On the other hand this gives a quick starting point when someone comes here first time without the need to find the right places in the examples or the docs. 
I would prefer to keep the snippets here bit I\`m OK with removing them as well.

Reply:
> That is really not critical. I'm actually more concerned about the API changes. User having to do `h3::client::new(conn, Default::default())` (the default part) kinda bugs me a little. Do you have any input on that?

Hmm good question, it is possible to call `h3::client::new(conn, ClientConfig::default())`. Another option is to make a `build` method for the Config structs which calls the `new` function, but this brings the older builder like style back.

All in all I do not have much experience in api design so I am not sure what is best here.